### PR TITLE
Fix the way datetime picker start time limits are computed

### PR DESCRIPTION
**Pull request by tomasr8**
fixes: https://github.com/orgs/indico/projects/7/views/1?pane=issue&itemId=145109248

Previously we weren't taking into account the current day in our limit calculation. For example, the `minStartDt` was being
applied to *all* days even though it should only apply if the currently selected date is the *same* as `minStartDt`. Same for `maxEndDt`.

Also adds the dynamic limit computation to session blocks which did not have it before.

### CODE REVIEW

File: indico/web/client/js/react/forms/fields.jsx [hunk 1]
```diff
@@ -429,11 +429,56 @@ function DateTimePickerComponent({
...
+      disabledMinutes = disabledMinutes.filter(
+        m => hour < minStartDt.hour() || (hour === minStartDt.hour() && m < minStartDt.minute())
+      );
+    }
+    if (shouldFilterEnd) {
+      disabledMinutes = disabledMinutes.filter(
+        m => hour > maxEndDt.hour() || (hour === maxEndDt.hour() && m > maxEndDt.minute())
+      );
+    }
+    return disabledMinutes;
```
Comment (AjobK) [inline indico/web/client/js/react/forms/fields.jsx]: Instead of filtering afterwards etc, I am wondering if you can just never add these keys to the list from the get-go. You basically take the difference between the start and end limit, then take that number and add x amount (based on start) to each minute so it doesn't start from `0` or `1`. Not essential, but I just figured this could maybe be slightly nicer?

Comment (tomasr8) [reply indico/web/client/js/react/forms/fields.jsx]: Yeah maybe doing it the way you suggest makes more sense, I'll think about it :+1:

File: indico/web/client/js/utils/date.ts [hunk 2]
```diff
@@ -5,8 +5,11 @@
 // modify it under the terms of the MIT License; see the
 // LICENSE file for more details.
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
```
Comment (tomasr8) [inline indico/web/client/js/utils/date.ts:9]: I will remove this in a follow up PR. There are a bunch of unrelated TS errors so I prefer to keep the fixes separate

File: indico/web/client/js/react/forms/fields.jsx (hunk 1 above)
Comment (tomasr8) [reply indico/web/client/js/react/forms/fields.jsx]: I simplified the functions and added tests since I actually overlooked a couple edge cases. Would you mind having another look?

File: indico/web/client/js/utils/__tests__/date.test.js [hunk 3]
```diff
@@ -110,3 +112,143 @@ describe('OpenDateRange', () => {
...
+    const result = getDisabledHours(dt, minStartDt);
+    expect(result).toEqual([]);
+  });
+
+  it('handles edge case: maxEndDt at midnight (hour 0)', () => {
+    const maxEndDt = moment('2025-01-15 00:00');
+    const result = getDisabledHours(dt, undefined, maxEndDt);
+    expect(result).toEqual([
+      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23,
+    ]);
```
Comment (AjobK) [inline indico/web/client/js/utils/__tests__/date.test.js:158]: Not really important but thought you might find it useful or interesting. If you prefer putting the explicit numbers like this it's also fine ofc
```suggestion
    expect(result).toEqual(Array(24).fill().map((_, i) => i+1));
```
